mypy/argmap.py

```diff
@@ -232,9 +243,9 @@ def expand_actual_type(
                     )
                     item = fallback.args[0]
                 return item
-            elif isinstance(actual_type, ParamSpecType):
+            elif isinstance(star_args_type, ParamSpecType):
```

Comment:
Should we check that it has the right flavor here?

Reply:
`ParamSpecType` has three flavors: bare, args and kwargs. Should this cond branch check that `star_args_type` is a `*args` flavor and not one of two remaining?

Reply:
(to be clear, that's as simple as `star_args_type.flavor == ParamSpecFlavor.ARGS`, it's already a enum-like thing, and not a real enum just to make mypyc happier)

Reply:
Would `mypyc` support something like this:

```python
class ParamSpecFlavor(IntEnum):
    BARE = 0
    ARGS = 1
    KWARGS = 2
    
class ParamSpecType[Flavor: ParamSpecFlavor](TypeVarLikeType): ...
```

Then we can annotate the return as `ParamSpecType[ParamSpecFlavor.ARGS]`